jina/serve/runtimes/gateway/request_handling.py

```diff
@@ -25,6 +26,9 @@ def _handle_request(request: 'Request') -> 'asyncio.Future':
 
         request_graph = copy.deepcopy(graph)
         # important that the gateway needs to have an instance of the graph per request
+        request_doc_ids = request.data.docs[
```

Comment:
We need to detect if there is a condition in the `graph` before doing it.

Doing serialization is rather `expensive` and I would rather avoid this if not needed